File: VB/RuntimeDbChooser.Module/ChangeDatabaseActiveDirectoryAuthentication.vb
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.Security.Strategy
Imports RuntimeDbChooser.Module.BusinessObjects

Public Class ChangeDatabaseActiveDirectoryAuthentication
    Inherits AuthenticationActiveDirectory(Of SecuritySystemUser, CustomLogonParametersForActiveDirectoryAuthentication)

    Public Sub New()
        CreateUserAutomatically = True
    End Sub

    Public Overrides ReadOnly Property IsLogoffEnabled As Boolean
        Get
            Return True
        End Get
    End Property

    Public Overrides ReadOnly Property AskLogonParametersViaUI As Boolean
        Get
            Return True
        End Get
    End Property
End Class

File: CS/RuntimeDbChooser.Module/ChangeDatabaseActiveDirectoryAuthentication.cs
#if !DXCORE3 && !NETSTANDARD
using DevExpress.ExpressApp.Security;
using DevExpress.ExpressApp.Security.Strategy;
using RuntimeDbChooser.Module.BusinessObjects;

namespace RuntimeDbChooser.Module {
    public class ChangeDatabaseActiveDirectoryAuthentication : AuthenticationActiveDirectory<SecuritySystemUser, CustomLogonParametersForActiveDirectoryAuthentication> {
        public ChangeDatabaseActiveDirectoryAuthentication() {
            CreateUserAutomatically = true;
        }
        public override bool IsLogoffEnabled {
            get {
                return true;
            }
        }
        public override bool AskLogonParametersViaUI {
            get {
                return true;
            }
        }
    }
}
#endif
